### ISSUE TITLE
Remove and Sort Usings on save Doesn't Work

### ISSUE THREAD

**Issue by Mike-EEE** [Microsoft/VS-PPT#81]
I have the latest version installed, but when I save my files, this doesn't seem to work as expected.  Is there another consideration here that I am missing?

**Comment by renebentes** [comment]
I'm having this problem also in .Net Core projects. No using is removed when saving.

**Comment by paulh-rnd** [comment]
Me too, in .Net Core only.

**Comment by ChrisHimsworth** [comment]
I have the same issue, but with non .NET Core and having ReSharper installed/enabled.

**Comment by VeselovAndrey** [comment]
Same for me. It was on RC and now on VS2017 RTM. Code formatting on save works but extension not updating "using" at all.

**Comment by andreydevua** [comment]
Same for me. with non .NET Core and having ReSharper installed/enabled.

**Comment by n-develop** [comment]
Same here. Only in VS 2017. ASP.NET Project (not Core). Also having Resharper installed.

**Comment by THammond9** [comment]
See [this stackoverflow question](http://stackoverflow.com/questions/644011/visual-studio-warning-level-meanings) for a solution to this problem that worked for me. I also posted in #40 with a little more detail on the problem.

**Comment by LukaszRabiec** [comment]
Same as @n-develop. In VS 2017 with Resharper.

**Comment by ChristopherHaws** [comment]
I have the same issue when Resharper is enabled. If I suspend Resharper, this feature works fine.

**Comment by xShivan** [comment]
I've got the same problem as the rest: non-.net core projects and Resharper installed - VS 2017 RTM.

**Comment by UweKeim** [comment]
Any update on this, whether this will/can ever be corrected?

**Comment by justcla** [comment]
@UweKeim Which issue are you referring to?
- Remove and Sort Usings On Save not working with .Net CORE projects
- Remove and Sort Usings on Save not working when ReSharper is installed?

**Comment by UweKeim** [comment]
Remove and Sort Usings on Save not working when ReSharper is installed.

**Comment by justcla** [comment]
That's really a bit out of our control.

Does it work when you uninstall ReSharper?
If so.... you can uninstall ReSharper. Or raise the issue with JetBrains.

**Comment by VeselovAndrey** [comment]
> If so.... you can uninstall ReSharper.
Are you serious? R# provides more features than "sort and remove". I will choose R# in this case.

>  Or raise the issue with JetBrains.
Using same logic JetBrains should return question back to you. Because R# works fine and they have no bug in their software.

**Comment by UweKeim** [comment]
I totally agree with @VeselovAndrey on this. Both parties will likely blame it on each other leaving us users with no solution.

Anyway, I've just reported it:

https://resharper-support.jetbrains.com/hc/en-us/community/posts/115000159930

Maybe someone is still willing to actually solve it.

**Comment by justcla** [comment]
@VeselovAndrey Well, I was partly seriously. If two extensions conflict with each other, you sometimes have to chose one over the other. I completely agree that R# offers far more feateures than a few Power Commands. Until it's resolved by us or JetBrains, I guess the Remove and Sort Usings on Save feature won't be functional if you're running R#. I'm happy to work with them to resolve our conflict.

@UweKeim That's fantastic! Thanks very much for raising the issue with JetBrains. Hopefully we'll get some resolution.

**Comment by VeselovAndrey** [comment]
@justcla IMHO investigation should starts from Power Commands. And when you'll find why its failing and got some proof (or ideas) how it conflicts with R# then go to JetBrains and raise an bug there. And provide them all details. BTW I completely understand that resolving such bugs are not easy and may require cooperation with JetBrains but it definitely should starts on Power Commands side.

Anyway, thanks for you and all other contributors work. Power Commands is great extensions and helps a lot in my job. And I hope you will resolve this bug soon. :)

**Comment by justcla** [comment]
OK. I'll try to get around to investigating the conflict. Maybe @olegtk can find the problem. :-)

**Comment by olegtk** [comment]
We should really open source Power Commands asap. Let me quickly check what's going on.

**Comment by Mike-EEE** [comment]
> If so.... you can uninstall ReSharper. 

Srsly? 😆

**Comment by justcla** [comment]
@Mike-EEE Just providing options. ;-)

**Comment by olegtk** [comment]
ok, fixing .net core issue is easy, installing resharper...

**Comment by olegtk** [comment]
cannot repro it, works fine with the latest Resharper on my machine.

**Comment by UweKeim** [comment]
Maybe it is a "red herring" that ReSharper is the cause?

I'll try setting up a fresh VMware and try to see whether I can repro this with/without Resharper.

(I'm using the Community Edition of Visual Studio 2017).

**Comment by VeselovAndrey** [comment]
@olegtk I have VS2017 with latest updates and latest R# (download and installed yesterday). Extension didn't work.

**Comment by Mike-EEE** [comment]
Update from [JetBrains' side](https://resharper-support.jetbrains.com/hc/en-us/community/posts/115000159930-VS-Productivity-Power-Tools-Remove-and-Sort-Usings-on-Save-not-working-with-R-?page=1#community_comment_115000202364):

>Please could you try unticking that checkbox "ReSharper | Options | Environment | Keyboard & Menus | Hide overridden VS menu items" and then check "Remove and Sort Usings on Save" feature once again.

**Comment by UweKeim** [comment]
That solution proposed by R# guy works for me!

> Please could you try unticking that checkbox "ReSharper | Options | Environment | Keyboard & Menus | Hide overridden VS menu items" and then check "Remove and Sort Usings on Save" feature once again.

**Comment by VeselovAndrey** [comment]
I get menu item back, it sorts but not remove unused "using".

**Comment by olegtk** [comment]
Yeah, I think that's good enough workaround. We could probably still make it work even with Resharper hiding the command, but it would be kinda hacky.

**Comment by UweKeim** [comment]
The solution proposed by R# guy does not work anymore for me with R# 2018.1. 😢
